fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1953,7 +1989,160 @@ public IndexingPolicy build() {
+            return new Builder();
+        }
+
+        /**
+         * Builder for {@link ScrubbingPolicy}.
+         *
+         * <pre><code>
+         * OnlineIndexer.ScrubbingPolicy.newBuilder().setReportMissingLimit(100).setScrubDangling(false).build()
+         * </code></pre>
+         *
+         */
+        @API(API.Status.UNSTABLE)
+        public static class Builder {
+            boolean scrubDangling = true;
+            boolean scrubMissing = true;
+            int reportDanglingLimit = 1000;
+            int reportMissingLimit = 1000;
+            boolean allowRepair = true;
+            long quota = 0;
+
+            protected Builder() {
+            }
+
+            /**
+             * If never called, the defaulted to allow scanning the index, looking for dangling index entries.
+             * @param scrubDangling if true
+             * @return this builder.
+             */
+            public Builder setScrubDangling(final boolean scrubDangling) {
+                this.scrubDangling = scrubDangling;
+                return this;
+            }
+
+            /**
+             * If never called, the defaulted to allow scanning the record store, looking for missing index entries.
+             * @param scrubMissing if true
+             * @return this builder.
+             */
+            public Builder setScrubMissing(final boolean scrubMissing) {
+                this.scrubMissing = scrubMissing;
+                return this;
+            }
+
+            /**
+             * If never called, the default is allowing (up to) 1000 error reports.
+             * @param reportDanglingLimit the max number of dangling-indexes errors to report.
+             * @return this builder.
+             */
+            public Builder setReportDanglingLimit(final int reportDanglingLimit) {
+                this.reportDanglingLimit = reportDanglingLimit;
+                return this;
+            }
+
+            /**
+             * If never called, the default is allowing (up to) 1000 error reports.
+             * @param reportMissingLimit the max number of missing-indexes errors to report.
+             * @return this builder.
+             */
+            public Builder setReportMissingLimit(final int reportMissingLimit) {
+                this.reportMissingLimit = reportMissingLimit;
+                return this;
+            }
+
+            /**
+             * Set 'scrub but do not repair' mode.
+             * @param val - if false, always report errors but do not repair.
+             *            - if true (default), report errors (up to report limits) but always repair.
+             * @return this builder.
+             */
+            public Builder setAllowRepair(boolean val) {
+                this.allowRepair = val;
+                return this;
+            }
+
+            /**
+             * set records/indexes scan quota. The scrubbing will return after scanning this quota, or more, records/indexes
```

Comment:
```suggestion
             * Set records/indexes scan quota. The scrubbing will return after scanning this quota, or more, records/indexes
```

I don't love the word "quota" here. Would "scrubbed record limit" maybe be better?

Reply:
"record limit" might be a little misleading. How about `entriesScannedLimit`?

Reply:
Changed to `entriesScanLimit`.